src/app_service/service/contacts/service.nim

```diff
@@ -330,6 +333,9 @@ QtObject:
       self.addContact(result)
 
   proc getStatusForContactWithId*(self: Service, publicKey: string): StatusUpdateDto =
+    if publicKey == singletonInstance.userProfile.getPubKey():
```

Comment:
@jrainville this will return default created `StatusUpdateDto` which won't be correct state in case somebody calls `getStatusForContactWithId` for logged in user. If crash is happening here in this proc, that means in some cases passed `publicKey` parameter is actually a public key of logged in user, if you agree let's return the correct `StatusUpdateDto` even in this case, at first look I see at least 2 approaches:
- to add in `then` block of this condition, the following (but that requires to `settingsService` assignment to `contactsService` instance):
```
if publicKey == singletonInstance.userProfile.getPubKey():
  let currentUserStatus = self.settingsService.getCurrentUserStatus()
  return StatusUpdateDto(publicKey: singletonInstance.userProfile.getPubKey(),
                                            statusType: currentUserStatus.statusType,
                                            clock: currentUserStatus.clock.uint64,
                                            text: currentUserStatus.text)
```
- to add entry for logged in user to the `contactsStatus` table from `init` proc on the app start and keep updating the table accordingly from `emitCurrentUserStatusChanged` proc, that way we will have always an accurate state in the table for logged in user, and this proc (`getStatusForContactWithId`) won't need code you added and won't crash.

Maybe the second variant is easier to implement, with this improvement we will be sure that `getStatusForContactWithId` proc will return current status for any user.